"""
This unit test checks that the argument checking for
sendmess() is sane. This is a librepy test.
"""
#pragma repy restrictions.default dylink.r2py librepy.r2py

# Try some invalid host args
try:
  sent = sendmess(123, 80, "Test")
  log("Sent mess! Bad host arg!\n")

except RepyArgumentError:
  pass

# Try bad ports
try:
  sent = sendmess("google.com", 3.14, "Test")
  log("Sent mess! Bad port arg!\n")

except RepyArgumentError:
  pass

# Try bad localip, should get binding error
try:
  sent = sendmess("google.com", 80, "Test", "123.213.111.222")
  log("Sent mess! Bad local ip!\n")

except AddressBindingError:
  pass

# Try an invalid IP, argument error
try:
  sent = sendmess("google.com", 80, "Test", 123)
  log("Sent mess! Invalid local ip!\n")

except RepyArgumentError:
  pass

# Try a bad local port, we don't have access
try:
  sent = sendmess("google.com", 80, "Test", "127.0.0.1", 80)
  log("Sent mess! Bad port!\n")

except ResourceForbiddenError:
  pass

# Try an invalid local port
try:
  sent = sendmess("google.com", 80, "Test", "127.0.0.1", 3.14)
  log("Sent mess! Invalid port!\n")

except RepyArgumentError:
  pass

